services/alertmanager/alertmanager_test.go

```diff
@@ -36,3 +44,95 @@ func TestAlertmanager(t *testing.T) {
 
 	require.NoError(t, svc.IsReady(context.Background()))
 }
+
+func TestPopulateConfig(t *testing.T) {
+	var cfg alertmanager.Config
+	buf, err := ioutil.ReadFile(alertmanagerBaseConfigPath)
+	require.NoError(t, err)
+	err = yaml.Unmarshal(buf, &cfg)
+	require.NoError(t, err)
+
+	sqlDB := testdb.Open(t, models.SkipFixtures, nil)
+	db := reform.NewDB(sqlDB, postgresql.Dialect, reform.NewPrintfLogger(t.Logf))
+	tx, err := db.Begin()
+	require.NoError(t, err)
+	defer func() {
+		require.NoError(t, tx.Rollback())
+	}()
+
+	q := tx.Querier
+
+	channel1, err := models.CreateChannel(q, &models.CreateChannelParams{
+		Summary: "some summary",
+		EmailConfig: &models.EmailConfig{
+			To: []string{"test@test.test"},
+		},
+		Disabled: false,
+	})
+	require.NoError(t, err)
+
+	channel2, err := models.CreateChannel(q, &models.CreateChannelParams{
+		Summary: "some summary",
+		PagerDutyConfig: &models.PagerDutyConfig{
+			RoutingKey: "ms-pagerduty-dev",
+		},
+		Disabled: false,
+	})
+	require.NoError(t, err)
+
+	templateName := gofakeit.UUID()
+	_, err = models.CreateTemplate(q, &models.CreateTemplateParams{
+		Template: &alert.Template{
+			Name:    templateName,
+			Version: 1,
+			Summary: gofakeit.Quote(),
+			Tiers:   []common.Tier{common.Anonymous},
+			Expr:    gofakeit.Quote(),
+			Params: []alert.Parameter{{
+				Name:    gofakeit.UUID(),
+				Summary: gofakeit.Quote(),
+				Unit:    gofakeit.Letter(),
+				Type:    alert.Float,
+				Range:   []interface{}{float64(10), float64(100)},
+				Value:   float64(50),
+			}},
+			For:         3,
+			Severity:    common.Warning,
+			Labels:      map[string]string{"foo": "bar"},
+			Annotations: nil,
+		},
+		Source: "USER_FILE",
+	})
+	require.NoError(t, err)
+
+	rule, err := models.CreateRule(q, &models.CreateRuleParams{
+		TemplateName: templateName,
+		Disabled:     true,
+		RuleParams: []models.RuleParam{
+			{
+				Name:       "test",
+				Type:       models.Float,
+				FloatValue: 3.14,
+			},
+		},
+		For:          5 * time.Second,
+		Severity:     common.Warning,
+		CustomLabels: map[string]string{"foo": "bar"},
+		Filters:      []models.Filter{{Type: models.Equal, Key: "value", Val: "10"}},
+		ChannelIDs:   []string{channel1.ID, channel2.ID},
+	})
+	require.NoError(t, err)
+
+	svc := New(db)
+	err = svc.populateConfig(&cfg)
+	require.NoError(t, err)
+
+	assert.Len(t, cfg.Receivers, 1)
+	assert.Equal(t, "empty", cfg.Receivers[0].Name)
+	assert.Equal(t, channel1.ID+" + "+channel2.ID, cfg.Receivers[1].Name)
```

Comment:
Two lines above you check that `cfg.Receivers` contains one element with index 0; here the second element is accessed

Reply:
thats a typo, that was 2 before. it is supposed to have 2 elements but for some reason the db transactions are failing while creating the rules in the testdb, which is why this assertion also fails.